### ISSUE TITLE
Add service name dimension to trace metrics

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-collector#857]
Add service name (`Node.ServiceInfo.Name`) dimension to receiver metrics. This helps to [determine](https://www.jaegertracing.io/docs/1.17/troubleshooting/#check-the-metrics-endpoint) which services are reporting spans.

The service label could be applied to trace receiver metrics but also to exporter metrics, which verifies that spans for particular services were exported.

Receiver metrics:
```
# HELP otelcol_receiver_accepted_spans Number of spans successfully pushed into the pipeline.
otelcol_receiver_accepted_spans{receiver="jaeger",transport="agent"} 3
# HELP otelcol_receiver_refused_spans Number of spans that could not be pushed into the pipeline.
otelcol_receiver_refused_spans{receiver="jaeger",transport="agent"} 0
```

Exporter metrics:
```
# HELP otelcol_exporter_send_failed_spans Number of spans in failed attempts to send to destination.
otelcol_exporter_send_failed_spans{exporter="jaeger"} 10
# HELP otelcol_exporter_sent_spans Number of spans successfully sent to destination.
otelcol_exporter_sent_spans{exporter="jaeger"} 3
```

**Comment by tigrannajaryan** [comment]
@pjanotti thoughts on this?

**Comment by pavolloffay** [comment]
@pjanotti any update on this one?

**Comment by bogdandrutu** [comment]
I think this adds significant overhead (potentially) and probably should be controlled by the level of telemetry user asks for.

**Comment by bogdandrutu** [comment]
Closing as a "sub-task" of https://github.com/open-telemetry/opentelemetry-collector/issues/2070

**Comment by ericmustin** [comment]
Hello.

This issue was closed and linked to https://github.com/open-telemetry/opentelemetry-collector/issues/2070 , however https://github.com/open-telemetry/opentelemetry-collector/issues/2070 doesn't appear to be implemented as far as I can tell, and was auto closed by the gh-actions stale bot.

Is there still interest in taking a PR on this, with the caveats mentioned (opt in telemetry level setting)?

**Comment by ericmustin** [comment]
I was able to get a workaround rigged up for my use case by combining the countconnector and the resource detection processor

receivers:
  foo:
exporters:
  bar:
connectors:
  count:
processors:
resourcedetection/system:
detectors: ["system"]
   system:
     hostname_sources: ["os"]
service:
  pipelines:
    traces:
      receivers: [foo]      
exporters: [count]    
metrics:
      receivers: [count]
processors: [resourcedetection/system]
exporters: [bar]


Simple example I got working for my use case. Not exactly what I wanted but, it works for my rough needs (just need to know which apps => which collectors).

Happy to join sig and chat a bit about whether there's a way to simplify this or make it easier for users.
